src/OpenSage.Game/Data/Apt/Geometry.cs

```diff
@@ -17,13 +43,273 @@ public static Geometry FromFileSystemEntry(FileSystemEntry entry)
             using (var reader = new StreamReader(stream))
             {
                 string line;
-                while ((line=reader.ReadLine())!=null)
+                int image = 0;
+                float thickness = 0;
+                Matrix2x2 rotMat = new Matrix2x2();
+                Vector2 translation = new Vector2();
+                ColorRgba color = new ColorRgba(0, 0, 0, 0);
+                GeometryStyle style = GeometryStyle.Undefined;
+                var tris = new List<Triangle2D>();
+                var lines = new List<Line2D>();
+
+                Action ApplyStyle = () =>
+                {
+                    switch (style)
+                    {
+                        //no data parsed yet
+                        case GeometryStyle.Undefined:
+                            break;
+                        case GeometryStyle.TexturedTri:
+                            geometry.Entries.Add(new GeometryTexturedTriangles(tris, color,image,rotMat,translation));
+                            tris.Clear();
+                            break;
+                        case GeometryStyle.SolidTri:
+                            geometry.Entries.Add(new GeometrySolidTriangles(tris, color));
```

Comment:
I think it would be cleaner to pass `new List<Triangle2D>(tris)` here, rather than doing it in the `GeometrySolidTriangles` constructor.

Reply:
Will do